src/ParallelAlgorithms/LinearSolver/AsynchronousSolvers/ElementActions.hpp

```diff
@@ -184,7 +184,7 @@ struct PrepareSolve {
         make_not_null(&box),
         [](const gsl::not_null<size_t*> iteration_id,
            const gsl::not_null<double*> residual_magnitude_square,
-           const db::const_item_type<residual_tag>& residual) noexcept {
+           const typename residual_tag::type& residual) noexcept {
```

Comment:
In #2306 these `const_item_type` are replaced by `const auto&`. That's related to an earlier comment that I think we should decide how to handle this kind of thing and then be consistent about it.